pkg/registry/core/serviceaccount/storage/token.go

```diff
@@ -62,30 +64,75 @@ var gvk = schema.GroupVersionKind{
 }
 
 func (r *TokenREST) Create(ctx context.Context, name string, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
-	if createValidation != nil {
-		if err := createValidation(ctx, obj.DeepCopyObject()); err != nil {
-			return nil, err
-		}
-	}
+	req := obj.(*authenticationapi.TokenRequest)
 
-	out := obj.(*authenticationapi.TokenRequest)
+	// Get the namespace from the context (populated from the URL).
+	namespace, ok := genericapirequest.NamespaceFrom(ctx)
+	if !ok {
+		return nil, errors.NewBadRequest("namespace is required")
+	}
 
-	if errs := authenticationvalidation.ValidateTokenRequest(out); len(errs) != 0 {
-		return nil, errors.NewInvalid(gvk.GroupKind(), "", errs)
+	// require name/namespace in the body to match URL if specified
+	if len(req.Name) > 0 && req.Name != name {
+		errs := field.ErrorList{field.Invalid(field.NewPath("metadata").Child("name"), req.Name, "must match the service account name if specified")}
+		return nil, errors.NewInvalid(gvk.GroupKind(), name, errs)
+	}
+	if len(req.Namespace) > 0 && req.Namespace != namespace {
+		errs := field.ErrorList{field.Invalid(field.NewPath("metadata").Child("namespace"), req.Namespace, "must match the service account namespace if specified")}
+		return nil, errors.NewInvalid(gvk.GroupKind(), name, errs)
 	}
 
+	// Lookup service account
 	svcacctObj, err := r.svcaccts.Get(ctx, name, &metav1.GetOptions{})
 	if err != nil {
 		return nil, err
 	}
 	svcacct := svcacctObj.(*api.ServiceAccount)
 
+	// honor uid as a precondition if set
+	if len(req.UID) > 0 && req.UID != svcacct.UID {
```

Comment:
submitting a Binding object to pods/binding uses the uid / resourceVersion of the submitted binding object (if set) as a precondition, requiring it to match the parent pod's uid and resourceVersion

Reply:
I can buy that those are transformations of a subset of the data in the parent object, rather than a distinct object like TokenRequest. I can back out the uid bits from this if desired.

Reply:
> that's not relevant to the kubectl cmd, so I'm happy to split that bit out for discussion

Yes, let's split it out.  The rest of the PR lgtm.

Reply:
@sftim exactly the issue we are facing, we don't have the `UID` at admission time and it makes it hard to correlate the admission request with the corresponding resource.